src/gluonnlp/model/bert.py

```diff
@@ -30,13 +30,212 @@
+        in_bias = F.concat(query_bias, key_bias, value_bias, dim=0)
+        in_weight = F.concat(query_weight, key_weight, value_weight, dim=0)
+        # qkv_proj shape = (seq_length, batch_size, num_heads * head_dim * 3)
+        qkv_proj = F.FullyConnected(data=qkv, weight=in_weight, bias=in_bias,
+                                    num_hidden=self.units*3, no_bias=False, flatten=False)
+        att_score = F.contrib.interleaved_matmul_selfatt_qk(qkv_proj, heads=self._num_heads)
+        if valid_len is not None:
+            valid_len = F.broadcast_axis(F.expand_dims(valid_len, axis=1),
+                                         axis=1, size=self._num_heads)
+            valid_len = valid_len.reshape(shape=(-1, 0), reverse=True)
+            att_weights = F.softmax(att_score, length=valid_len, use_length=True, axis=-1)
+        else:
+            att_weights = F.softmax(att_score, axis=-1)
+        # att_weights shape = (batch_size, seq_length, seq_length)
+        att_weights = self.dropout_layer(att_weights)
+        context_vec = F.contrib.interleaved_matmul_selfatt_valatt(qkv_proj, att_weights,
+                                                                  heads=self._num_heads)
+        att_weights = att_weights.reshape(shape=(-1, self._num_heads, 0, 0), reverse=True)
+        return context_vec, att_weights
+
+
+class BERTEncoderCell(HybridBlock):
+    """Structure of the BERT Encoder Cell.
+
+    Parameters
+    ----------
+    units : int
+        Number of units for the output
+    hidden_size : int
+        number of units in the hidden layer of position-wise feed-forward networks
+    num_heads : int
+        Number of heads in multi-head attention
+    dropout : float
+    output_attention: bool
+        Whether to output the attention weights
+    weight_initializer : str or Initializer
+        Initializer for the input weights matrix, used for the linear
+        transformation of the inputs.
+    bias_initializer : str or Initializer
+        Initializer for the bias vector.
+    prefix : str, default None
+        Prefix for name of `Block`s. (and name of weight if params is `None`).
+    params : Parameter or None
+        Container for weight sharing between cells. Created if `None`.
+    activation : str, default 'gelu'
+        Activation methods in PositionwiseFFN
+    layer_norm_eps : float, default 1e-5
+        Epsilon for layer_norm
+
+    Inputs:
+        - **inputs** : input sequence. Shape (length, batch_size, C_in)
+        - **valid_length** : valid length of inputs for attention. Shape (batch_size, length)
+
+    Outputs:
+        - **outputs**: output tensor of the transformer encoder cell.
+            Shape (length, batch_size, C_out)
+        - **additional_outputs**: the additional output of all the BERT encoder cell.
+    """
+    def __init__(self, units=128, hidden_size=512, num_heads=4,
+                 dropout=0.0, output_attention=False,
```

Comment:
`use_bias : bool` is missing.

Reply:
We didn't have this flag in the previous version. https://github.com/dmlc/gluon-nlp/blob/v0.8.x/src/gluonnlp/model/bert.py#L198 
Is it necessary to add this flag?

Reply:
As attention has its flag, so I think it is better to add it.

Reply:
added